util/base/src/test/java/jetbrains/jetpad/base/AsyncsTest.java

```diff
@@ -23,45 +23,72 @@
 import java.io.IOException;
 import java.util.Arrays;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
+import static jetbrains.jetpad.base.AsyncMatchers.failed;
+import static jetbrains.jetpad.base.AsyncMatchers.result;
+import static jetbrains.jetpad.base.AsyncMatchers.voidSuccess;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
 import static org.junit.Assert.assertSame;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertThat;
 import static org.junit.Assert.fail;
 
 public class AsyncsTest {
   @Test
   public void constantAsync() {
-    Async<Integer> c = Asyncs.constant(239);
-    assertSuccess(c, 239);
+    assertThat(Asyncs.constant(239), result(equalTo(239)));
   }
 
   @Test
   public void failureAsync() {
-    assertFailure(Asyncs.<Integer>failure(new Throwable()));
+    assertThat(Asyncs.<Integer>failure(new Throwable()), AsyncMatchers.<Integer>failed());
   }
 
   @Test
-  public void map() {
+  public void internalMap() {
     Async<Integer> c = Asyncs.constant(239);
     Async<Integer> mapped = Asyncs.map(c, new Function<Integer, Integer>() {
       @Override
       public Integer apply(Integer input) {
         return input + 1;
       }
     }, new SimpleAsync<Integer>());
-    assertSuccess(mapped, 240);
+    assertThat(mapped, result(equalTo(240)));
   }
 
   @Test
-  public void mapFailure() {
+  public void map() {
+    Async<Integer> c = Asyncs.constant(239);
+    Async<Integer> mapped = c.map(new Function<Integer, Integer>() {
+      @Override
+      public Integer apply(Integer input) {
+        return input + 1;
+      }
+    });
+    assertThat(mapped, result(equalTo(240)));
+  }
+
+  @Test
+  public void internalMapFailure() {
     Async<Integer> failure = Asyncs.failure(new Throwable());
-    assertFailure(Asyncs.map(failure, new Function<Integer, Integer>() {
+    Async<Integer> mapped = Asyncs.map(failure, new Function<Integer, Integer>() {
       @Override
       public Integer apply(Integer input) {
         return input + 1;
       }
-    }, new SimpleAsync<Integer>()));
+    }, new SimpleAsync<Integer>());
+    assertThat(mapped, AsyncMatchers.<Integer>failed());
```

Comment:
As far as I understand, we can get rid of explicit type parameters is such calls